cmd/cloudFoundryCreateService.go

```diff
@@ -0,0 +1,103 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/SAP/jenkins-library/pkg/cloudfoundry"
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/pkg/errors"
+)
+
+func cloudFoundryCreateService(config cloudFoundryCreateServiceOptions, telemetryData *telemetry.CustomData) {
+
+	cf := cloudfoundry.CFUtils{Exec: &command.Command{}}
+
+	var c = cf.Exec
+
+	//var c command.ExecRunner = &command.Command{}
+
+	c.Stdout(log.Entry().Writer())
+	c.Stderr(log.Entry().Writer())
+
+	var err, logouterr error
+
+	loginOptions := cloudfoundry.LoginOptions{
+		CfAPIEndpoint: config.CfAPIEndpoint,
+		CfOrg:         config.CfOrg,
+		CfSpace:       config.CfSpace,
+		Username:      config.Username,
+		Password:      config.Password,
+	}
+
+	err = cf.Login(loginOptions)
+	if err == nil {
+		err = runCloudFoundryCreateService(&config, telemetryData, c)
+	}
+	if err != nil {
+		logouterr = cf.Logout()
+		if logouterr != nil {
+			log.Entry().WithError(logouterr).Fatal("step execution failed")
+		}
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+	logouterr = cf.Logout()
+	if logouterr != nil {
+		log.Entry().WithError(logouterr).Fatal("step execution failed")
+	}
+	log.Entry().Info("Service creation completed successfully")
+}
+
+func runCloudFoundryCreateService(config *cloudFoundryCreateServiceOptions, telemetryData *telemetry.CustomData, c command.ExecRunner) error {
+	var err error
+	log.Entry().Info("Creating Cloud Foundry Service")
+
+	var cfCreateServiceScript []string
+	cfCreateServiceScript = []string{"create-service", config.CfService, config.CfServicePlan, config.CfServiceInstanceName}
+
+	if config.CfServiceBroker != "" {
+		cfCreateServiceScript = append(cfCreateServiceScript, "-b", config.CfServiceBroker)
+	}
+	if config.CfCreateServiceConfig != "" {
+		cfCreateServiceScript = append(cfCreateServiceScript, "-c", config.CfCreateServiceConfig)
+	}
+	if config.CfServiceTags != "" {
+		cfCreateServiceScript = append(cfCreateServiceScript, "-t", config.CfServiceTags)
+	}
+	if config.ServiceManifest != "" {
+		var varPart []string
+		cfCreateServiceScript = []string{"create-service-push", "--no-push", "--service-manifest", config.ServiceManifest}
+
+		if len(config.ManifestVariablesFiles) >= 0 {
+			for _, v := range config.ManifestVariablesFiles {
+				if fileExists(v) {
+					cfCreateServiceScript = append(cfCreateServiceScript, "--vars-file", v)
```

Comment:
[just sidenote] I have also to deal with `vars-file`s in cfDeploy (currently another PR). I would like to suggest to continue for now simply, but to refactor later when both PRs are merged in order to avoid duplications ....

Reply:
Sure, let's have a talk on that! Functionality should be the same for both steps, right?